From 597cd63b48c6dc3746827504f191744c120fbb4d Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 10 Sep 2021 21:10:03 +0100
Subject: [PATCH 0898/2201] gpio-fsm: Clamp the delay time to zero

The sysfs delay_ms value is calculated live, and it is possible for
the time left to appear to be negative briefly if the timer handling
hasn't completed. Ensure the displayed value never goes below zero,
for the sake of appearances.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/gpio/gpio-fsm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-fsm.c b/drivers/gpio/gpio-fsm.c
index 306f5123546c..3a348f1c6514 100644
--- a/drivers/gpio/gpio-fsm.c
+++ b/drivers/gpio/gpio-fsm.c
@@ -884,7 +884,7 @@ static ssize_t delay_ms_show(struct device *dev,
 	const struct gpio_fsm *gf = dev_get_drvdata(dev);
 	int jiffies_left;
 
-	jiffies_left = gf->delay_jiffies - jiffies;
+	jiffies_left = max((int)(gf->delay_jiffies - jiffies), 0);
 	return sprintf(buf,
 		       gf->delay_target_state ? "%u\n" : "-\n",
 		       jiffies_to_msecs(jiffies_left));
-- 
2.17.1

